src/common/utxobased/engine/makeUtxoEngineState.ts

```diff
@@ -736,55 +731,42 @@ export const pickNextTask = async (
   }
 
   // Check if there are any addresses pending to be subscribed
-  if (
-    Object.keys(addressSubscribeCache).length > 0 &&
-    !taskCache.addressWatching
-  ) {
+  if (Object.keys(addressSubscribeCache).length > 0) {
+    // These are addresse to which the server has not subscribed
+    const newAddress: string[] = []
     const blockHeight = serverStates.getBlockHeight(uri)
-    // Loop each address that needs to be subscribed
+
+    // Loop each address in the cache
     for (const [address, state] of Object.entries(addressSubscribeCache)) {
       // Add address in the cache to the set of addresses to watch
-      const { path, processing: subscribed } = state
-      // only process newly watched addresses
-      if (subscribed) continue
-      if (path != null) {
-        // Add the newly watched addresses to the UTXO cache
-        addressUtxoCache[address] = {
-          processing: false,
-          path
-        }
-        await addToAddressTransactionCache(
-          args,
-          address,
-          path.format,
-          path.changeIndex,
-          blockHeight,
-          addressTransactionCache
-        )
+      if (!serverStates.serverIsAwareOfAddress(uri, address)) {
+        newAddress.push(address)
       }
+
+      // Add to the addressTransactionCache if they're not yet added:
+      // Only process newly watched addresses
+      if (state.processing) continue
+      // Add the newly watched addresses to the UTXO cache
+      addressUtxoCache[address] = {
+        processing: false,
```

Comment:
Why do we even need this processing now?
Is it used somewhere else in the code that's not part of the commit?

Since it was used before on line 747:

```js 
const { path, processing: subscribed } = state
// only process newly watched addresses
if (subscribed) continue
```

but now it's not used (at least in the code here) anymore.

Reply:
the processing flag on the `addressSubscribedCache` entry is to signal whether we have added an entry to `addressUtxoCache` already.